platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc

```diff
@@ -147,6 +147,16 @@ When this happens, configurations in the substituted project could be selected t
 
 Consuming non-consumable configurations in this manner is deprecated and will result in an error in Gradle 9.0.
 
+[[variants_with_no_artifacts]]
+=== Defining secondary variants for primary variants does not remove primary variants
+
+It is possible to define [Secondary Variants](https://docs.gradle.org/8.9/userguide/publishing_customization.html#sec:publishing-custom-components) for consumable "primary" variants that lack artifacts.
+
+Defining secondary variants for a primary variant lacking artifacts used to result in the primary variant being removed from consideration during variant selection.
```

Comment:
```suggestion
In earlier versions of Gradle, doing so would cause the primary variant to be silently excluded from variant selection, because it lacked artifacts.
```